mpitest/test_mpi_opt.py

```diff
@@ -162,44 +170,290 @@ def test_parab_subbed_Pcomps(self):
+class ParallelMPIOptAsym(MPITestCase):
+    N_PROCS = 2
+
+    def setUp(self):
+        prob = Problem(impl=impl)
+        root = prob.root = Group()
+        #root.ln_solver = PetscKSP()  # this works too
+        root.ln_solver = LinearGaussSeidel()
+        par = root.add('par', ParallelGroup())
+
+        ser1 = par.add('ser1', Group())
+
+        ser1.add('p1', ParamComp('x', np.zeros([2])), promotes=['*'])
+        ser1.add('comp', SimpleArrayComp(), promotes=['*'])
+        ser1.add('con', ExecComp('c = y - 20.0', c=np.array([0.0, 0.0]),
+                                  y=np.array([0.0, 0.0])), promotes=['*'])
+        ser1.add('obj', ExecComp('o = y[0]', y=np.array([0.0, 0.0])),
+                                 promotes=['*'])
+
+        ser2 = par.add('ser2', Group())
+        ser2.add('p1', ParamComp('x', np.zeros([2])), promotes=['*'])
+        ser2.add('comp', SimpleArrayComp(), promotes=['*'])
+        ser2.add('obj', ExecComp('o = y[0]', y=np.array([0.0, 0.0])),
+                                  promotes=['*'])
+
+        root.add('con', ExecComp('c = y - 30.0', c=np.array([0.0, 0.0]),
+                                 y=np.array([0.0, 0.0])))
+        root.add('total', ExecComp('obj = x1 + x2'))
+
+        root.connect('par.ser1.o', 'total.x1')
+        root.connect('par.ser2.o', 'total.x2')
+        root.connect('par.ser2.y', 'con.y')
+
+        prob.driver = pyOptSparseDriver()
+        prob.driver.add_param('par.ser1.x', low=-50.0, high=50.0)
+        prob.driver.add_param('par.ser2.x', low=-50.0, high=50.0)
+
+        prob.driver.add_objective('total.obj')
+        prob.driver.add_constraint('par.ser1.c', ctype='eq')
+        prob.driver.add_constraint('con.c', ctype='eq')
+
+        self.prob = prob
+
+    def tearDown(self):
+        try:
+            os.remove('SNOPT_print.out')
+            os.remove('SNOPT_summary.out')
+        except OSError:
+            pass
+
+    def test_parallel_array_comps_asym_fwd(self):
+        prob = self.prob
+        prob.root.ln_solver.options['mode'] = 'fwd'
+        prob.root.par.ln_solver.options['mode'] = 'fwd'
+        prob.root.par.ser1.ln_solver.options['mode'] = 'fwd'
+        prob.root.par.ser2.ln_solver.options['mode'] = 'fwd'
+
+        prob.setup(check=False)
+        prob.run()
+
+        assert_rel_error(self, prob['total.obj'], 50.0, 1e-6)
+
+    def test_parallel_array_comps_asym_rev(self):
+        prob = self.prob
+        prob.root.ln_solver.options['mode'] = 'rev'
+        prob.root.par.ln_solver.options['mode'] = 'rev'
+        prob.root.par.ser1.ln_solver.options['mode'] = 'rev'
+        prob.root.par.ser2.ln_solver.options['mode'] = 'rev'
+
+        prob.setup(check=False)
+        prob.run()
+
+        assert_rel_error(self, prob['total.obj'], 50.0, 1e-6)
+
+class ParallelMPIOptPromoted(MPITestCase):
+    N_PROCS = 2
+
+    def setUp(self):
+        prob = Problem(impl=impl)
+        root = prob.root = Group()
```

Comment:
Dont need, this or the next 3 lines, since they get over ridden by the assignment of a new root glass. 

If you meant to have lin_gs as the lienar solver, need to move it down below the assignment of the parallelGroup as root